packages/lesswrong/components/ea-forum/users/EAGApplicationImportForm.tsx

```diff
@@ -0,0 +1,702 @@
+    howOthersCanHelpMeRef?.current?.setEditorValue(importedData.howOthersCanHelpMe.markdownValue)
+    howICanHelpOthersRef?.current?.setEditorValue(importedData.howICanHelpOthers.markdownValue)
+  }
+  
+  const handleCopyField = async (e, field) => {
+    e.preventDefault()
+    if (!importedData) return
+
+    if (field === 'biography') {
+      biographyRef?.current?.setEditorValue(importedData.biography.markdownValue)
+      return
+    } else if (field === 'howOthersCanHelpMe') {
+      howOthersCanHelpMeRef?.current?.setEditorValue(importedData.howOthersCanHelpMe.markdownValue)
+      return
+    } else if (field === 'howICanHelpOthers') {
+      howICanHelpOthersRef?.current?.setEditorValue(importedData.howICanHelpOthers.markdownValue)
+      return
+    // } else if (field === 'organizerOfGroupIds') {
+    //   // TODO: this field needs more work -
+    //   // prob need to only list groups they are assigned to on the forum, and link to the form for requests
+    //   setFormValues({
+    //     ...formValues,
+    //     [field]: importedData.organizerOfGroups.map(g => g._id)
+    //   })
+    //   return
+    } else if (field === 'mapLocation') {
+      const location = await importMapLocation()
+      if (location) {
+        setFormValues({
+          ...formValues,
+          mapLocation: location
+        })
+      }
+      return
+    } else if (field === 'linkedinProfileURL') {
+      setFormValues({
+        ...formValues,
+        linkedinProfileURL: importLinkedinProfileURL()
+      })
+      return
+    }
+    
+    setFormValues({
+      ...formValues,
+      [field]: importedData[field]
+    })
+  }
+  
+  const handleChangeField = (e, field) => {
+    e.preventDefault()
+    setFormValues({
+      ...formValues,
+      [field]: e.target.value
+    })
+  }
+  
+  const handleUpdateValue = (val) => {
+    setFormValues({
+      ...formValues,
+      ...val
+    })
+  }
+  
+  const handleSubmit = async (e, copyAll=false) => {
+    e.preventDefault()
+    
+    let updatedFormData = {...formValues}
+    if (copyAll && importedData) {
+      // @ts-ignore
+      updatedFormData = {
+        ...importedData,
+        mapLocation: await importMapLocation(),
+        linkedinProfileURL: importLinkedinProfileURL()
+      }
+      // update CKEditor fields
+      biographyRef?.current?.setEditorValue(importedData.biography.markdownValue)
+      howOthersCanHelpMeRef?.current?.setEditorValue(importedData.howOthersCanHelpMe.markdownValue)
+      howICanHelpOthersRef?.current?.setEditorValue(importedData.howICanHelpOthers.markdownValue)
+    }
+    
+    for (let field in updatedFormData) {
+      if (Array.isArray(updatedFormData[field]) && !updatedFormData[field].length) {
+        updatedFormData[field] = null
+      }
+    }
+    
+    // add data from CKEditor fields
+    const submission = submitFormCallbacks.current.reduce((prev, next) => {
+      try {
+        return next(prev)
+      } catch (e) {
```

Comment:
Add a comment about why we are silently ignoring errors?

Reply:
Yeah, I decided to actually let the user know and prevent submission if there's an error here. BTW thanks for all your code review comments, I found them all really helpful! :)